src/dmd/expressionsem.d

```diff
@@ -11397,6 +11397,25 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return false;
         }
 
+        if (sc && sc.flags & SCOPE.Cfile)
```

Comment:
While this works for the specific case of A==B, shouldn't this go in as part of the general implicit conversion code?

Reply:
#13049.